src/ui/ui_settings/ui_settings.js

```diff
@@ -0,0 +1,123 @@
+import { defaultsDeep } from 'lodash';
+import Bluebird from 'bluebird';
+
+import { getDefaultSettings } from './defaults';
+
+function hydrateUserSettings(user) {
+  return Object.keys(user).reduce(expand, {});
+  function expand(expanded, key) {
+    const userValue = user[key];
+    if (userValue !== null) {
+      expanded[key] = { userValue };
+    }
+    return expanded;
+  }
+}
+
+function assertRequest(req) {
+  if (
+    typeof req === 'object' &&
+    typeof req.path === 'string' &&
+    typeof req.headers === 'object'
+  ) return;
+
+  throw new TypeError('all uiSettings methods must be passed a hapi.Request object');
+}
+
+export class UiSettings {
+  constructor(server, status) {
+    this._server = server;
+    this._status = status;
+  }
+
+  getDefaults() {
+    return getDefaultSettings();
+  }
+
+  // returns a Promise for the value of the requested setting
+  async get(req, key) {
+    assertRequest(req);
+    return this.getAll(req)
+      .then(all => all[key]);
+  }
+
+  async getAll(req) {
+    assertRequest(req);
+    return this.getRaw(req)
+      .then(raw => Object.keys(raw)
+        .reduce((all, key) => {
+          const item = raw[key];
+          const hasUserValue = 'userValue' in item;
+          all[key] = hasUserValue ? item.userValue : item.value;
+          return all;
+        }, {})
+      );
+  }
+
+  async getRaw(req) {
+    assertRequest(req);
+    return Promise
```

Comment:
__nit:__ since `getDefaults()` doesn't return `Promise` anymore, maybe it will make sense to get rid of `Promise.all` and use something like this instead?
```
return this.getUserProvided(req)
      .then((user) => defaultsDeep(user, this.getDefaults()));
```